src/components/SplitCsvUpload/SplitCsvUpload.tsx

```diff
@@ -0,0 +1,104 @@
+import { UploadOutlined } from '@ant-design/icons'
+import { Trans } from '@lingui/macro'
+import TooltipIcon from 'components/TooltipIcon'
+import { ThemeContext } from 'contexts/themeContext'
+import { Split } from 'models/v2/splits'
+import { ChangeEventHandler, useContext } from 'react'
+import { splitPercentFrom } from 'utils/v2/math'
+
+// https://developer.mozilla.org/en-US/docs/Web/API/FileReader/readyState
+const READY_STATE_DONE = 2
+
+const readCsvFile = async (file: File): Promise<string | null> => {
+  const reader = new FileReader()
+
+  return new Promise((resolve, reject) => {
+    reader.onload = function (evt) {
+      if (evt?.target?.readyState != READY_STATE_DONE) return resolve(null)
+      if (evt?.target.error) {
+        reject(evt.target.error)
+        return
+      }
+
+      const content = evt.target.result as string | null
+
+      resolve(content)
+    }
+
+    reader.readAsText(file)
+  })
+}
+
+const parseCsvFile = (csvContent: string) => {
+  const [, ...rows] = csvContent.split('\n')
+
+  const splits: Split[] = rows.map(row => {
+    const [
+      beneficiary,
+      percent,
+      preferClaimed,
+      lockedUntil,
+      projectId,
+      allocator,
+    ] = row.split(',')
+
+    return {
+      beneficiary,
+      percent: splitPercentFrom(parseFloat(percent) * 100).toNumber(),
+      preferClaimed: Boolean(preferClaimed),
+      lockedUntil: lockedUntil ? parseInt(lockedUntil) : undefined,
+      projectId: projectId || undefined,
+      allocator: allocator || undefined,
+    }
+  })
+
+  return splits
+}
+
+export function SplitCsvUpload({
+  onChange,
+}: {
+  onChange: (splits: Split[]) => void
+}) {
+  const {
+    theme: { colors },
+  } = useContext(ThemeContext)
+
+  const onUploadChange: ChangeEventHandler<HTMLInputElement> = async e => {
+    e.preventDefault()
+    e.stopPropagation()
+
+    const { files } = e.target
+    const file = files?.[0]
+    if (!file) return
+
+    const csvContent = await readCsvFile(file)
+    if (!csvContent) return
```

Comment:
is it worth adding some kind of error state to show the user with these if null: return conditions?

Reply:
Yeah I was a bit fast-n-loose on the error handling.

Went thru and handled any unhandled errors with error noti's:

Example of empty file.

<img width="517" alt="Screen Shot 2022-08-09 at 12 07 59 PM" src="https://user-images.githubusercontent.com/12551741/183551719-a6648aee-4b73-4a9c-aea1-3171e0f1058b.png">